api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,85 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig configures the extension that runs the endpoint picking service.
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
```

Comment:
Include `,omitempty` in the json tag for optional fields.

Reply:
@ahg-g This field is still missing `,omitempty`. Additionally, The `+kubebuilder:validation:Required` marker is intended to ensure that a field is always present in API requests so this field should not be apointer.

Reply:
The pointer is to make sure we have a path in the future to relax the validation here when adding other types, recall this is a union. I added the omitempty.